src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -531,7 +531,9 @@ void CascadiaSettings::_validateMediaResources()
         // Explicitly just use the Icon here, not the EvaluatedIcon. We don't
         // want to blow up if we fell back to the commandline and the
         // commandline _isn't an icon_.
-        if (const auto icon = profile.Icon(); icon.size() > 2)
+        // GH #17943: "none" is a special value interpreted as "remove the icon"
+        static hstring HideIconValue{ L"none" };

```

Comment:
```suggestion
        static constexpr std::wstring_view HideIconValue{ L"none" };
```

we should always prefer these - hstrings need to be constructed, and that construction takes place when the DLL gets loaded or when this function is called for the first time.